src/structures/paging/frame.rs

```diff
@@ -133,7 +133,7 @@ impl<S: PageSize> Sub<PhysFrame<S>> for PhysFrame<S> {
 }
 
 /// An range of physical memory frames, exclusive the upper bound.
-#[derive(Clone, Copy, PartialEq, Eq)]
+#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
```

Comment:
Unfortunately there is no reasonable ordering for range types. For example, should `11..15` be considered smaller or larger than `12..13`? Deriving `PartialOrd` here would order by the start bound first, which would make `1..999999` smaller than `2..3`, which can be surprising.

See also the [range type](https://doc.rust-lang.org/stable/std/ops/struct.Range.html) in Rust's standard library, which does not implement `PartialOrd` either.

Deriving `Hash` for range types seems fine.

Reply:
My initial reasoning with this was there often isn't an obvious ordering that makes sense, but still providing one anyway could still be useful eg to store them as keys in a `BTreeMap` where the meaning of the ordering doesn't need to have any meaning but there has to be an ordering.

In hindsight `Range` not implementing is a reasonable argument for not implementing `PartialOrd`/`Ord` that to avoid confusion.

Reply:
Yeah, for `BTreeMap` it doesn't really matter, but `PartialOrd`/`Ord` impls are also used in other places, e.g. in [`sort_unstable`](https://doc.rust-lang.org/stable/std/primitive.slice.html#method.sort_unstable) or [`Iterator::max`](https://doc.rust-lang.org/stable/core/iter/trait.Iterator.html#method.max), where the ordering does matter. So I think it's better to implement the traits only if there really is some logical ordering between the elements. For storing things in a `BTreeMap` users can still add a custom `OrdWrapper` type with an arbitrary `Ord` implementation.